tutorials/09-set-up-physics-engine-tpe.md

```diff
@@ -0,0 +1,246 @@
+> { };
+
+class EntityManagementFeatures :
+  public virtual Base,
+  public virtual Implements3d<EntityManagementFeatureList>
+{
+  ... // other features' member functions
+
+  // ----- Construct empty entities -----
+  public: Identity ConstructEmptyWorld(
+    const Identity &_engineID, const std::string &_name) override;
+
+  ... // other features' member functions
+};
+
+}
+}
+}
+```
+
+Together with other \ref ignition::physics::Feature "Features", the \ref ignition::physics::ConstructEmptyWorldFeature "ConstructEmptyWorldFeature"
+is included in `EntityManagementFeatureList` "FeatureList" to declare the related
+features for entity management purpose.
+
+The `EntityManagementFeatures` "FeatureList" here inherits from:
+- [Base](https://github.com/ignitionrobotics/ign-physics/blob/main/tpe/plugin/src/Base.hh)
+class for foundation metadata definitions of Models, Joints, Links, and Shapes objects
+of TPE to provide easy access to [tpelib](https://github.com/ignitionrobotics/ign-physics/tree/main/tpe/lib)
+structures in the TPE library.
+- \ref ignition::physics::Implements3d "Implements3d" for implementing the
+custom feature with \ref ignition::physics::FeaturePolicy3d "FeaturePolicy3d"
+("FeaturePolicy" of 3 dimensions and scalar type `double`).
+
+Based on each \ref ignition::physics::Feature "Features" class definition, we
+will override the corresponding functions of all the class defined in
+`EntityManagementFeatureList` (the feature definitions can be found [here](https://ignitionrobotics.org/api/physics/3.0/classes.html)).
+Here we show the overriding of `ConstructEmptyWorld` member function of \ref ignition::physics::ConstructEmptyWorldFeature "ConstructEmptyWorldFeature".
+
+- [EntityManagementFeatures.cc](https://github.com/ignitionrobotics/ign-physics/blob/main/tpe/plugin/src/EntityManagementFeatures.cc):
+
+```cpp
+#include <string>
+
+#include "EntityManagementFeatures.hh"
+
+using namespace ignition;
+using namespace physics;
+using namespace tpeplugin;
+
+... // other function implementations
+
+/////////////////////////////////////////////////
+Identity EntityManagementFeatures::ConstructEmptyWorld(
+  const Identity &, const std::string &_name)
+{
+  auto world = std::make_shared<tpelib::World>();
+  world->SetName(_name);
+  return this->AddWorld(world);
+}
+
+... // other function implementations
+```
+
+To realize the \ref ignition::physics::ConstructEmptyWorldFeature "ConstructEmptyWorldFeature"
+on the TPE physics engine, we implement `ConstructEmptyWorld` method. We simply
+instantiate \ref ignition::physics::tpelib::World "World" object and then
+wrap it with `std::share_ptr`. We then set the corresponding input world name
+and store the current live world object.
+
+Please take a look at [EntityManagement_TEST.cc](https://github.com/ignitionrobotics/ign-physics/blob/main/tpe/plugin/src/EntityManagement_TEST.cc)
+as an example unit test for this `EntityManagementFeatures` "FeatureList".
+
+### Main plugin.cc file
```

Comment:
I think this section should be placed before `Implement a feature`, cuz this should be the first step to create a skeleton folder structure. This is when they design the `ign-physics` plugin and think about what features are needed.